src/Common/src/CoreLib/Interop/Unix/System.Native/Interop.GetTimestamp.cs

```diff
@@ -8,10 +8,10 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestampResolution")]
-        internal static extern bool GetTimestampResolution(out long resolution);
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestampResolution", BestFitMapping = false, ExactSpelling = true)]
+        internal static extern unsafe BOOL GetTimestampResolution(long* resolution);
```

Comment:
The makes the C# definition to not match unmanaged implementation. If you would like to start using `BOOL` in the Unix shims, the following should happen:

- The `BOOL` should be moved from Windows specific directory to platform neutral place
- The C shim should be changed to return BOOL in .h/c.

`BOOL` on Unix looks foreign to me, but I do not have strong opinion about this. @stephentoub Do you like `BOOL` in Unix shims?

Reply:
> BOOL on Unix looks foreign to me, but I do not have strong opinion about this. @stephentoub Do you like BOOL in Unix shims?

I'd prefer just using `int`.

Reply:
We will have some disconnect somewhere. We have a managed wrapper method that matches the Windows signature on both. On UNIX that then P/Invoke calls into a SystemNative wrapper method that returns `int` and itself calls the appropriate native method. On Windows it is a direct P/Invoke call that returns `BOOL` (which is just an `int` typedef), so we need to normalize this difference somewhere.

Reply:
I think just using `int` everywhere makes the most sense. It also avoids people assuming `TRUE == 1`, when it is actually `TRUE != 0`.

Reply:
> We have a managed wrapper method that matches the Windows signature on both

This match is accidental. We do generally factor the code to take advantage of similar accidental matches.

Reply:
> We will have some disconnect somewhere.

The P/Invokes should map 1:1 wherever possible with the native export they call to.  If platform-agnostic higher-level code needs to call these P/Invokes that differ in signature, it should have a platform-specific PAL method in the middle that hides the differences.

Reply:
> The P/Invokes should map 1:1 wherever possible with the native export they call to

The code is overall simpler by making the Unix and Windows signatures match here. I only changed it to `BOOL` originally (for Windows) at the request of @jkotas. I then opted to update Unix as well to make the code overall simpler.

I think just using `int` is probably the simplest path to make the signatures match and keep the code simple. This is being touched because it was cropping up for IOT scenarios and is measurable in regular apps that are heavily dependent on stopwatch so I don't think adding more layers of abstraction is worthwhile.

Given that `BOOL` is just a typedef and not a user-defined struct, using `int` is still matching the underlying signature. That is, ignoring the header file, the underlying signature (as exported from the library) is: `int QueryPerformanceFrequency(LARGE_INTEGER* pFrequency)`.

Reply:
I think it would be best to:
- Change `Stopwatch.QueryPerformanceCounter` to return `long` and not propagate the error condition.
- Change `Stopwatch.QueryPerformanceFrequency` to return `long` as well and remove the support for non-high performance timers. This will make the whole thing another bit faster. From MSDN docs for `QueryPerformanceFrequency`:  On systems that run Windows XP or later, the function will always succeed and will thus never return zero.

Reply:
@jkotas, it should be updated now.

The wrapper `Stopwatch.QueryPerformanceCount/Frequency` methods both return `long`.
On Unix, the `SystemNative_GetTimestamp[Resolution]` methods directly return `uint64_t` and assert the result code is as expected.
On Windows, the P/Invoke wrapper asserts the result code is as expected and returns the result directly.

The support for the low-resolution fallback timer was also removed.